auth/MobileController.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const rp = require('request-promise');
+const scoutuser = require('../scout_user');
+const url = require('url');
+
+var pocketConsumerKey, userAuthKey;
+
+const oathRequestOptions = {
+  uri: 'https://getpocket.com/v3/oauth/request',
+  method: 'POST',
+  body: '',
+  headers: {
+    'Content-Type': 'application/json; charset=UTF-8',
+    'X-Accept': 'application/json'
+  }
+};
+
+const finalAuthorizeOptions = {
+  uri: 'https://getpocket.com/v3/oauth/authorize',
+  method: 'POST',
+  body: '',
+  headers: {
+    'Content-Type': 'application/json; charset=UTF-8',
+    'X-Accept': 'application/json'
+  }
+};
+
+// Read the configuration file for pocket info.
+pocketConsumerKey = process.env.POCKET_KEY;
+
+//
+// Pocket Auth Flows
+//
+router.get('/login', function(req, res) {
+  const requrl = url.format({
+    protocol: req.protocol,
+    host: req.get('host')
+  });
+  const redirUri = `${requrl}/api/auth/mobile/redirecturi`;
+  console.log(`redirUri = ${redirUri}`);
+
+  var oauthBody = {
+    consumer_key: pocketConsumerKey,
+    redirect_uri: redirUri
+  };
+  oathRequestOptions.body = JSON.stringify(oauthBody);
+  rp(oathRequestOptions).then(function(body) {
+    let jsonBody = JSON.parse(body);
+    userAuthKey = jsonBody.code;
+
+    var redir =
+      `https://getpocket.com/auth/authorize` +
+      `?request_token=${userAuthKey}&redirect_uri=${redirUri}`;
+    return res.redirect(redir);
+  });
+});
+
+router.get('/redirecturi', function(req, res) {
+  var authBody = {
+    consumer_key: pocketConsumerKey,
+    code: userAuthKey
+  };
+  finalAuthorizeOptions.body = JSON.stringify(authBody);
+  console.log('calling redirect');
+
+  rp(finalAuthorizeOptions)
+    .then(function(body) {
+      let jsonBody = JSON.parse(body);
+      const pocketUserAccessToken = jsonBody.access_token;
+      const pockerUserId = jsonBody.username;
+
+      // Save to the config to scoutuser data
+      console.log(`Authorized: ${pockerUserId}/${pocketUserAccessToken}`);
+      processScoutUser(pockerUserId, pocketUserAccessToken);
+    })
+    .catch(function(err) {
+      console.log('Call failed' + err);
+    });
+  res.status(200).send('OK');
```

Comment:
This API is called by Pocket server, right?  So, I think we need to figure out how to get the token back to the mobile user so they can login with the token.  We can figure that out in another patch.  This looks great.